### DISCUSSION TITLE
Question on directus_refresh_token

### DISCUSSION THREAD

**Discussion by adanielyan** [directus/directus#10758] [Archive]
Hi, I am implementing Directus authentication in my Next.js app. I am able to get the directus_refresh_token using OAuth API endpoint.

What I don't understand, is why Directus only sends a refresh token but no access token when user is authenticated. I still have to get the access token using my refresh token, and I cannot think of a scenario when I won't need to do that right after authentication.

Am I missing something? What is the recommended auth flow that explains why only a refresh token is provided upon successful OAuth?

**Comment by rijkvanzanten** [comment]
> What I don't understand, is why Directus only sends a refresh token but no access token when user is authenticated. I still have to get the access token using my refresh token, and I cannot think of a scenario when I won't need to do that right after authentication.

The refresh token is set as a httpOnly cookie, meaning we can set it securely. However, seeing this token can't be read from client side JS, there's no way for Directus to safely send the access token during that initial redirect. The best way is to have the refresh token be set securely, and allow the user to refresh right after the fact 👍🏻 

(this does also have the added benefit of being able to call refresh on every first page load, meaning that you're previously logged-in users can stay logged in, and only having to show the login page w/ oauth links for users where that refresh call fails. After the redirect from oauth, the same page-loading logic kicks in, allowing the user to continue to the app directly. This is actually exactly how the Directus app itself works as well.)
